File: patch_api.js
const fs = require('fs');
const file = 'frontend/src/services/api.js';
let code = fs.readFileSync(file, 'utf-8');

code = code.replace(/getNearbyServices: \(id\) => api.get\(\`\/hotels\/\$\{id\}\/nearby-services\`\),/, "getNearbyServices: (id) => api.get(`/hotels/${id}/nearby-services`),\n  getPricePreview: (id, params) => api.get(`/hotels/${id}/price-preview`, { params }),");

fs.writeFileSync(file, code);
console.log("PATCHED API");

File: patch_api.py
import re

with open('frontend/src/services/api.js', 'r', encoding='utf-8') as f:
    content = f.read()

content = content.replace("getCompetitorBenchmarking: () => api.get('/manager/competitor-benchmarking'),", "getCompetitorBenchmarking: (params) => api.get('/manager/competitor-benchmarking', { params }),")

with open('frontend/src/services/api.js', 'w', encoding='utf-8') as f:
    f.write(content)
